ts/utils/login.ts

```diff
@@ -35,14 +36,24 @@ export const extractLoginResult = (url: string): LoginResult | undefined => {
   }
 
   // Check for LOGIN_FAILURE
-  const failureTokenPathPos = url.indexOf(LOGIN_FAILURE_PREFIX);
-  if (failureTokenPathPos !== -1) {
-    const errCode = url.substr(
-      failureTokenPathPos + LOGIN_FAILURE_PREFIX.length
+  if (url.indexOf(LOGIN_FAILURE_PREFIX) !== -1) {
+    const failureWithErrorCodeTokenPathPos = url.indexOf(
+      LOGIN_FAILURE_WITH_ERROR_CODE_PREFIX
     );
+    // try to extract error code
+    if (failureWithErrorCodeTokenPathPos !== -1) {
+      const errCode = url.substr(
+        failureWithErrorCodeTokenPathPos +
+          LOGIN_FAILURE_WITH_ERROR_CODE_PREFIX.length
+      );
+      return {
+        success: false,
+        errorCode: errCode.length > 0 ? errCode : undefined
```

Comment:
I see that the error code is returned to the IdpSelectionScreen. What about adding it as the payload of the `loginFailure` action? It could increase the error traking capability

Reply:
Totally agree